src/app.jsx

```diff
@@ -179,28 +170,25 @@
         this.onNavigate = () => this.setState({ path: cockpit.location.path });
     }
 
-    componentDidMount() {
+    async componentDidMount() {
         cockpit.addEventListener("locationchanged", this.onNavigate);
 
         if (this.props.error)
             this.onAddErrorNotification({ text: _("Failed to fetch some resources"), detail: this.props.error });
 
-        cockpit.script('type virt-install', { err: 'ignore' })
-                .then(() => {
-                    this.setState({ virtInstallAvailable: true });
-                    cockpit.spawn(['virt-install', '--install=?'], { err: 'ignore' })
-                            .then(() => this.setState({ downloadOSSupported: true }),
-                                  () => this.setState({ downloadOSSupported: false }));
-
-                    cockpit.spawn(['virt-install', '--cloud-init=?'], { err: 'ignore' })
-                            .then(() => this.setState({ cloudInitSupported: true }),
-                                  () => this.setState({ cloudInitSupported: false }));
-
-                    cockpit.spawn(['virt-install', '--unattended=?'], { err: 'ignore' })
-                            .then(options => this.setState({ unattendedSupported: true, unattendedUserLogin: options.includes('user-login') }),
-                                  () => this.setState({ unattendedSupported: false }));
-                },
-                      () => this.setState({ virtInstallAvailable: false }));
+        const check_exec = argv => cockpit.spawn(argv, { err: 'ignore' })
+                .catch(() => false);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1404-20240129-134001-6e94a503-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)